nav2_map_server/include/nav2_map_server/map_saver_core.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2020 Shivam Pandey
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_MAP_SERVER__MAP_SAVER_CORE_HPP_
+#define NAV2_MAP_SERVER__MAP_SAVER_CORE_HPP_
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+
+namespace nav2_map_server
+{
+
+/**
+ * @brief nav2_map_server::MapSaver class for savemap utils and services
+ * The class is templated with the map type and supports OccupancyGrids
+ * and Pointcloud based maps for now.
+ * @tparam mapT Template argument for Map type
+ */
+template<class mapT>
+class MapSaver : public nav2_util::LifecycleNode
```

Comment:
As far as I can tell, this file is never used - the MapSaver for 2D and 3D derive from the LifecycleNode, not this.